cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc

```diff
@@ -54,50 +73,93 @@ CALL db.indexes
-For this reason Cypher needs to know the property type in order to benefit from this feature.
-In the query above, the fact that we have a predicate comparing `p.name` to the String `'Tom'` is enough for Cypher to know the query is only interested in `String` properties, and we can skip the property lookup.
+For non-native indexes and the spatial type `Point`, there will still be a second DB access to retrieve those values. In indexes with mixed values,
+only those values that cannot be lookup up from the index will trigger another database access, while the other values will not.
 
 Note also that if you have pre-existing indexes that were build on lucene, upgrading to Neo4j 3.5 is _not sufficient_ to use the new index.
 It is necessary to drop and re-create the index in order to port it to the native index.
-Read more about this in the <<operations-manual#index-configuration-upgrade-considerations, Operations Manual -> Index Configuration Upgrade Considerations>>.
+For information on native index support see the _Operations Manual_:
 
-== Index-backed order-by
+* <<operations-manual#index-configuration-native-indexes, Property types supported by the native index>>
+* <<operations-manual#index-configuration-index-providers, Index providers and property types>>
+* <<operations-manual#index-configuration-upgrade-considerations, Index configuration upgrade considerations>>
+
+Predicates that can be used to enable this optimization are:
+
+* Existance (`WHERE exists(n.name)`)
+* Equality (e.g. `WHERE n.name = 'Tom Hanks'`)
+* Range (eg. `WHERE n.uid > 1000 AND n.uid < 2000`)
+* Prefix (eg. `WHERE n.name STARTS WITH 'Tom'`)
+* Suffix (eg. `WHERE n.name ENDS WITH 'Hanks'`)
+* Substring (eg. `WHERE n.name CONTAINS 'a'`)
+
+=== Index-backed order-by
 
 Now consider the following refinement to the query:
 
 [source, cypher]
 ----
-MATCH (p:Person)-[:ACTED_IN]->(m:Movie) USING INDEX p:Person(name) WHERE p.name STARTS WITH 'Tom'
+MATCH (p:Person)-[:ACTED_IN]->(m:Movie) WHERE p.name STARTS WITH 'Tom'
 RETURN p.name, count(m) ORDER BY p.name
 ----
 
 //table
 
 We are asking for the results in ascending alphabetical order. The new native index happens to store the String properties in ascending alphabetical order, and Cypher knows this.
-In Neo4j 3.4 and earlier, Cypher would plan a `Sort` operation to eagerly sort the results, which means building a collection in memory and running a sort algorithm on it.
-For large result sets this can be expensive in both memory and time. In Neo4j 3.5, if you are using the native index, Cypher will recognise that the index already returns data in the correct order, and skip the `Sort` operation.
+In Neo4j 3.4 and earlier, Cypher would plan a `Sort` operation to sort the results, which means building a collection in memory and running a sort algorithm on it.
+For large result sets this can be expensive in both memory and time. In Neo4j 3.5, if you are using the native index, Cypher will recognise that the index already
```

Comment:
```suggestion
For large result sets this can be expensive in terms of both memory and time. 
In Neo4j 3.5, if you are using the native index, Cypher will recognise that the index already
```